lib/Resque/Job.php

```diff
@@ -230,7 +230,12 @@ public function fail($exception)
 			$this->worker,
 			$this->queue
 		);
-		Resque_Stat::incr('failed');
+
+        if (Resque_Failure::getBackend() == 'Resque_Failure_RedisMultipleQueues') {
+            Resque_Stat::incr($this->queue . '_failed');
+        } else {
+            Resque_Stat::incr('failed');
+        }
```

Comment:
This fail method shouldn't need to know or care what type of failure object we are using.  Since ``$this->failure`` exists on the ``Resque_Job`` anyway, is there any reason we can't simply change this line to ``Resque_Stat::incr($this->queue . '_failed');`` and not have the conditional?